engine/verification/assigner/engine_test.go

```diff
@@ -374,12 +376,49 @@ func chunkQueueUnhappyPathDuplicate(t *testing.T) {
 	s.metrics.On("OnExecutionResultReceivedAtAssignerEngine").Return().Once()
 	e.ProcessFinalizedBlock(containerBlock)
 
+	unittest.RequireReturnsBefore(t, chunksQueueWG.Wait, 10*time.Millisecond, "could not receive chunk locators")
+
 	mock.AssertExpectationsForObjects(t,
 		s.metrics,
 		s.assigner,
-		s.chunksQueue,
 		s.notifier)
 
 	// job listener should not be notified as no new chunk is added.
 	s.newChunkListener.AssertNotCalled(t, "Check")
 }
+
+// mockChunksQueueForAssignment mocks chunks queue against invoking its store functionality for the
+// input assignment.
+// The mocked version of chunks queue evaluates that whatever chunk locator is tried to be stored belongs to the
+// assigned list of chunks for specified execution result (i.e., a valid input).
+// It also mocks the chunks queue to return the specified boolean and error values upon trying to store a valid input.
+func mockChunksQueueForAssignment(t *testing.T,
+	verId flow.Identifier,
+	chunksQueue *storage.ChunksQueue,
+	resultID flow.Identifier,
+	assignment *chunks.Assignment,
+	returnBool bool,
+	returnError error) *sync.WaitGroup {
+
+	wg := &sync.WaitGroup{}
+	wg.Add(len(assignment.ByNodeID(verId)))
+	chunksQueue.On("StoreChunkLocator", mock.Anything).Run(func(args mock.Arguments) {
+		// should be a chunk locator
+		locator, ok := args[0].(*chunks.Locator)
+		require.True(t, ok)
+
+		// should belong to the expected execution result and assigned chunk
+		require.Equal(t, resultID, locator.ResultID)
+		require.Contains(t, assignment.ByNodeID(verId), locator.Index)
+
+		wg.Done()
+	}).Return(
+		func(locator *chunks.Locator) bool {
+			return returnBool
+		},
+		func(locator *chunks.Locator) error {
+			return returnError
+		})
```

Comment:
```suggestion
	}).Return(returnBool, returnError)
```
Does the simpler format not work with multiple return values?